ballerina/context.bal

```diff
@@ -132,10 +132,16 @@ public isolated class Context {
                     return next;
                 }
             }
-            self.nextInterceptor = 0;
+            self.resetInterceptorCount();
             return;
         }
     }
+
+    isolated function resetInterceptorCount() {
+        lock {
+            self.nextInterceptor = 0;
+        }
+    }
```

Comment:
Will this solve the issue correctly? What happens when the parallel field execution?
Maybe we need to think about a way to handle this more elegantly.

Reply:
Earlier the problem was not resetting the interceptor counter when that interceptor is returning an error without executing the actual resolver. Don't we run fields sequentially for a request on the ballerina side? If so, since context is being created for every request, this won't be an issue. will it?